redis/cache/redispipebp/monitored.go

```diff
@@ -83,17 +111,43 @@ func (s MonitoredSync) Send(ctx context.Context, r redis.Request) (result interf
 		s.Name+".send",
 		tracing.SpanTypeOption{Type: tracing.SpanTypeClient},
 	)
+	command := extractCommand(r.Cmd)
+	if command == "" {
+		command = "send"
+	}
+	active := redisprom.ActiveRequests.With(prometheus.Labels{
+		redisprom.ClientNameLabel: s.Name,
+		redisprom.CommandLabel:    command,
+		redisprom.DatabaseLabel:   "", // We don't have that info
+		redisprom.TypeLabel:       "", // We don't have that info
+		redisprom.DeploymentLabel: "", // We don't have that info
+	})
+	active.Inc()
 	defer func(start time.Time) {
 		err := redis.AsError(result)
-		cmd := extractCommand(r.Cmd)
-		if cmd == "" {
-			cmd = "send"
-		}
+		durationSeconds := time.Since(start).Seconds()
 		promHistogram.With(prometheus.Labels{
 			labelSlug:    s.Name,
-			labelCommand: cmd,
+			labelCommand: command,
 			labelSuccess: prometheusbp.BoolString(err == nil),
-		}).Observe(time.Since(start).Seconds())
+		}).Observe(durationSeconds)
+		active.Dec()
```

Comment:
this should also be moved to be the first thing of the defer'd function, as [kyle's suggestion](https://github.com/reddit/baseplate.go/pull/552#discussion_r938997231)

Reply:
updated